azurerm/resource_arm_dns_ptr_record.go

```diff
@@ -0,0 +1,169 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/arm/dns"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmDnsPtrRecord() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDnsPtrRecordCreate,
+		Read:   resourceArmDnsPtrRecordRead,
+		Update: resourceArmDnsPtrRecordCreate,
+		Delete: resourceArmDnsPtrRecordDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"zone_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"records": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"ttl": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmDnsPtrRecordCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	dnsClient := client.dnsClient
+
+	name := d.Get("name").(string)
+	resGroup := d.Get("resource_group_name").(string)
+	zoneName := d.Get("zone_name").(string)
+	ttl := int64(d.Get("ttl").(int))
+	tags := d.Get("tags").(map[string]interface{})
+
+	metadata := expandTags(tags)
+
+	recordStrings := d.Get("records").(*schema.Set).List()
```

Comment:
we'd generally tend to pull this out into a separate function - could we update this to match?